### DISCUSSION TITLE
Faster image copy to other buffers

### DISCUSSION THREAD

**Discussion by lovettchris** [SixLabors/ImageSharp#1422] [Q&A] [answered]
### Description
I need to interoperate between ImageSharp and other libraries like UAP or WPF WriteableBitmap for reasons outside my control and so I would love a fast memcpy solution for whole image frame contiguous buffer copies.  Seems to me you could just expose the ImageFrame<T>.PixelBuffer and then expose GetSingleMemory, so System.Memory can do the CopyTo and we're good.

### Steps to Reproduce
Try create the quickest way to copy Image to WriteableBitmap, currently have to iterate 1 row at a time?

### System Configuration

- ImageSharp version: 1.0.1
- Other ImageSharp packages and versions: n/a
- Environment (Operating system, version and so on): Windows 10
- .NET Framework version: .net core 3.1 and soon 5.0...
- Additional information:

**Comment by JimBobSquarePants** [comment, marked as answer]
If the backing buffer is contiguous (_which is often the case unless dealing with multi-megapixel images_) we've got [`Image<TPixel>.TryGetSinglePixelSpan(out Span<TPixel>)`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image-1.html#SixLabors_ImageSharp_Image_1_TryGetSinglePixelSpan_System_Span__0___)

See https://github.com/SixLabors/ImageSharp/discussions/1375#discussioncomment-92897

**Reply by antonfirsov** [reply]
@lovettchris note that - in order to access the backing buffers - you need the generic `Image<TPixel>` and not the non-generic `Image` base type.

You can also create an "image" copy-free with [`Image.WrapMemory<TPixel>`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image.html#SixLabors_ImageSharp_Image_WrapMemory__1_SixLabors_ImageSharp_Configuration_System_Buffers_IMemoryOwner___0__System_Int32_System_Int32_). See the usage sample in #1419, would be nice to hear your thoughts on that proposal.

**Comment by lovettchris** [comment]
Thanks for the quick response, Image<TPixel>.TryGetSinglePixelSpan(out Span<TPixel>) works great.
